File: deploy/dux-front/lib/features/timetree/domain/models/timetree_audit_log.dart
class TimetreeAuditLog {
  final String id;
  final String? username;
  final String action;
  final String entityType;
  final String? entityId;
  final String result;
  final String? ipAddress;
  final DateTime actionDate;
  final String? details;

  const TimetreeAuditLog({
    required this.id,
    this.username,
    required this.action,
    required this.entityType,
    this.entityId,
    required this.result,
    this.ipAddress,
    required this.actionDate,
    this.details,
  });

  factory TimetreeAuditLog.fromJson(Map<String, dynamic> json) {
    return TimetreeAuditLog(
      id: (json['id'] ?? '').toString(),
      username: json['username'] as String?,
      action: json['action'] as String? ?? '',
      entityType: json['entityType'] as String? ?? '',
      entityId: json['entityId']?.toString(),
      result: json['result'] as String? ?? '',
      ipAddress: json['ipAddress'] as String?,
      actionDate: DateTime.parse(json['actionDate'] as String).toLocal(),
      details: json['details'] as String?,
    );
  }
}

File: deploy/dux-integration/src/main/java/com/asm/dux/timetree/domain/TimetreeAuditLog.java
package com.asm.dux.timetree.domain;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "TT_AUDIT_LOG", schema = "dbo")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class TimetreeAuditLog {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ID")
    private Long id;

    @Column(name = "USERNAME", length = 100)
    private String username;

    @Column(name = "ACTION", length = 100)
    private String action;

    @Column(name = "ENTITY_TYPE", length = 100)
    private String entityType;

    @Column(name = "ENTITY_ID")
    private Long entityId;

    @Column(name = "RESULT", length = 50)
    private String result;

    @Column(name = "IP_ADDRESS", length = 100)
    private String ipAddress;

    @Column(name = "ACTION_DATE")
    private LocalDateTime actionDate;

    @Lob
    @Column(name = "DETAILS", columnDefinition = "VARCHAR(MAX)")
    private String details;
}
